### PR TITLE
Bug/2042 2043 bugs in subscripitons payloads

### LINKED ISSUE #2042: Subscription list: the value of description is not returned
When I create a subscription with a description, I can see its description if I perform the request:
```
GET /v2/subscriptions/57108b3269aa5b493a556e07
{
  "id": "57108b3269aa5b493a556e07",
  "description": "One subscription to rule them all",
  "expires": "2016-04-05T14:00:00.00Z",
  "status": "expired",
  "subject": {
    "entities": [
      {
        "id": "",
        "idPattern": ".*",
        "type": "Room"
      }
    ],
    "condition": {
      "attributes": [
        "temperature"
      ],
      "expression": {
        "q": "temperature>40",
        "geometry": "",
        "coords": "",
        "georel": ""
      }
    }
  },
  "notification": {
    "callback": "http://localhost:1234",
    "throttling": 5,
    "attributes": [
      "humidity",
      "temperature"
    ]
  }
}
```
But when I list subscriptions, the description is not available:
```
GET /v2/subscriptions?limit=15&offset=0&options=count
[
  {
    "id": "57108b3269aa5b493a556e07",
    "expires": "2016-04-05T14:00:00.00Z",
    "status": "expired",
    "subject": {
      "entities": [
        {
          "id": "",
          "idPattern": ".*",
          "type": "Room"
        }
      ],
      "condition": {
        "attributes": [
          "temperature"
        ],
        "expression": {
          "q": "temperature>40",
          "geometry": "",
          "coords": "",
          "georel": ""
        }
      }
    },
    "notification": {
      "callback": "http://localhost:1234",
      "throttling": 5,
      "attributes": [
        "humidity",
        "temperature"
      ]
    }
  },
  {
    "id": "5710904a69aa5b493a556e08",
    "expires": "2016-04-05T14:00:00.00Z",
    "status": "expired",
    "subject": {
      "entities": [
        {
          "id": "",
          "idPattern": ".*",
          "type": "Car"
        }
      ],
      "condition": {
        "attributes": [
          "temperature"
        ],
        "expression": {
          "q": "temperature>40",
          "geometry": "",
          "coords": "",
          "georel": ""
        }
      }
    },
    "notification": {
      "callback": "http://localhost:1234",
      "throttling": 5,
      "attributes": [
        "humidity",
        "temperature"
      ]
    }
  }
]
```

### PR DESCRIPTION
**Pull request by fgalan**
Fixes #2042 and #2043 

@crbrox 
@kzangeli

### CODE REVIEW

File: src/lib/apiTypesV2/Subscription.cpp [hunk 1]
```diff
@@ -120,12 +120,34 @@ std::string Condition::toJson()
...
+  }
+  if (this->expression.georel != "")
+  {
+    atLeastOneSubField = true;
+    jhe.addString("georel", this->expression.georel);
+  }
+
+  if (atLeastOneSubField)
+  {
+    atLeastOneSubField = true;
```
Comment (kzangeli) [inline src/lib/apiTypesV2/Subscription.cpp]: This last 'atLeastOneSubField = true' doesn't seem necessary ...
Also, you could remove them all and replace with a **long** if statement:

    if ((expression.q != "") || (expression.geometry != "") || ...)
    {
      jh.addRaw("expression", jhe.str());
    }

Or better, perhaps:    if (jhe.str() != "") ?

Comment (fgalan) [inline src/lib/apiTypesV2/Subscription.cpp]: I like the second alternative. Test running right now...

Comment (fgalan) [inline src/lib/apiTypesV2/Subscription.cpp]: Fixed in https://github.com/telefonicaid/fiware-orion/pull/2050/commits/9d5e1b9a23fb59f8ab3232c9e81edd451da1a38f.

Note that the new approach allow to adopt a more "compact" coding style for this function.

**Comment by kzangeli** [comment]
LGTM

**Comment by crbrox** [comment]
LGTM

**Pull request merged**